vault_cli/vault_python_api.py

```diff
@@ -17,12 +17,12 @@
 def userpass_authentication(username, password):
     d = {"password": password}
     r = s.post(url + 'auth/userpass/login/' + username, json=d, headers={})
-    json_response = r.json()
 
     if r.status_code == requests.codes.ok:
+        json_response = r.json()
```

Comment:
Would `r.raise_for_status` help ?

Reply:
I think check_status should rather be called in vault.py

Reply:
We shoud also exit cleanly if no auth method has been specified

Reply:
> I think check_status should rather be called in vault.py

Updated: `check_status` is now used with every authentication methods.

> We shoud also exit cleanly if no auth method has been specified

Program exits cleanly if no authentication method has been specified, isn't it ?

Reply:
For now yes, but we only have authentication options, just need to keep it in mind

Reply:
> Would r.raise_for_status help ?

`r.ok` might be used but Vault API [documentation](https://www.vaultproject.io/api/auth/userpass/index.html#login) mentions only HTTP code `200` even if `400` if returned when authentication fails.